tools/gke-autopsc-controller/controllers/gateway_controller.go

```diff
@@ -0,0 +1,297 @@
+			logger.Info("change on `ConnectionPreference` detected")
+			change_detected = true
+		}
+
+		if !NatSubnetIsEqual(attachment.NatSubnets, newAttachment.NatSubnets) {
+			logger.Info("change on `NatSubnets` detected", "old", attachment.NatSubnets, "new", newAttachment.NatSubnets)
+			change_detected = true
+		}
+
+		if !reflect.DeepEqual(attachment.DomainNames, newAttachment.DomainNames) {
+			logger.Info("change on `DomainNames` detected", "old", attachment.DomainNames, "new", newAttachment.DomainNames)
+			change_detected = true
+		}
+
+		if !reflect.DeepEqual(attachment.ConsumerAcceptLists, newAttachment.ConsumerAcceptLists) {
+			logger.Info("change on `ConsumerAcceptLists` detected", "old", attachment.ConsumerAcceptLists, "new", newAttachment.ConsumerAcceptLists)
+			change_detected = true
+		}
+		if change_detected {
+			newAttachment.Fingerprint = attachment.Fingerprint
+			newAttachment.TargetService = ""
+			newAttachment.Name = ""
+			lop, err := service.Patch(r.project, r.region, name, newAttachment).Context(ctx).Do()
+			if err != nil {
+				logger.Error(err, "Failed updating ServiceAttachment")
+				return fmt.Errorf("failed_updating")
+			} else {
+				logger.Info("Updating ServiceAttachment")
+				go r.trackLopUpdate(ctx, req, gw, lop, logger)
+			}
+		} else {
+			logger.Info("No Change detected, no update necessary")
+		}
+	}
+	return nil
+}
+
+func createServieAttachmentFromGateway(gw *gatewayv1beta1.Gateway) *compute.ServiceAttachment {
+	var connectionPreference = "ACCEPT_AUTOMATIC"
+	consumerAcceptLists, connectionPreference := extractConnectionPreferenceAndConsumerAcceptList(gw, connectionPreference)
+	natSubnets := convertCommaSeparatedStringToArray(gw, "controller.autonpsc.dev/psc-serviceattachment-natsubnets")
+	domainNames := convertCommaSeparatedStringToArray(gw, "controller.autonpsc.dev/psc-serviceattachment-domainNames")
+	forwardingRuleName := gw.Annotations["networking.gke.io/forwarding-rules"]
+
+	gwToCreate := compute.ServiceAttachment{
+		ConnectionPreference: connectionPreference,
+		ConsumerAcceptLists:  consumerAcceptLists,
+		Name:                 gw.Annotations["controller.autonpsc.dev/psc-serviceattachment"],
+		//EnableProxyProtocol:  false,
+		NatSubnets:    natSubnets,
+		DomainNames:   domainNames,
+		TargetService: forwardingRuleName,
+	}
+
+	if gw.Annotations["controller.autonpsc.dev/psc-serviceattachment-domainNames"] == "" {
+		gwToCreate.DomainNames = nil
+	}
+
+	if gw.Annotations["controller.autonpsc.dev/psc-serviceattachment-natsubnets"] == "" {
+		gwToCreate.NatSubnets = nil
+	}
+
+	if gw.Annotations["controller.autonpsc.dev/psc-serviceattachment-allowed"] == "" {
+		gwToCreate.ConsumerAcceptLists = nil
+	}
+
+	return &gwToCreate
+}
+
+func convertCommaSeparatedStringToArray(gw *gatewayv1beta1.Gateway, propertyName string) []string {
+	var elements []string
+	if gw.Annotations[propertyName] != "" {
+		for _, element := range strings.Split(gw.Annotations[propertyName], ",") {
```

Comment:
just a suggestion, The following snippet of code removes the need for a `for` loop and improves readability:

`elements := strings.FieldsFunc(propertyValue, func(c rune) bool { return c == ',' })`

Reply:
That's a wonderful idea.